### ISSUE TITLE
Malformed URL

### ISSUE THREAD

**Issue by amgarr** [hippothomas/waistline-api#4]
I am getting an error when attempting to instantiate the database and reset the cache. Error is "Malformed URL". 

Not sure what the issue with my URL is. I defined it in env.local.

**Comment by hippothomas** [comment]
Hi! Could you give me more information about your environment and how did you setup the app?

If you are using docker compose, it may be an issue from the database url, the defaults one needs to be changed.
At least the host needs to change from the default value "127.0.0.1", like this :

```
DATABASE_URL="postgresql://app:!ChangeMe!@waistline-api_postgres:5432/waistline?serverVersion=16&charset=utf8"
MONGODB_URL="mongodb://root:!ChangeMe!@waistline-api_mongodb:27017"
```

It could also come from your database password, if you have a special character in it, you may need to encode it.

**Comment by amgarr** [comment]
Thanks, I had the host set up wrong.

Now I am getting an error with connecting to postgresql server. Logs are below. I dont have any special characters in the postgres or mongodb passwords. should I change from scram-sha-256?

I am setting it up using Docker compose. 

2024-02-28 16:06:07 2024-02-28 21:06:07.306 UTC [34] FATAL:  password authentication failed for user "app"
2024-02-28 16:06:07 2024-02-28 21:06:07.306 UTC [34] DETAIL:  Connection matched file "/var/lib/postgresql/data/pg_hba.conf" line 128: "host all all all scram-sha-256"

**Comment by hippothomas** [comment]
It may be because you changed the password after initializing the database, in this case you could delete the `data` folder and rerun `docker compose up -d`. It will recreate everything based on your new configuration.

**Comment by amgarr** [comment]
I have rerun docker compose and still getting the same issue.

**Comment by hippothomas** [comment]
This seems to be a configuration issue and I will need more details to be able to help you, can you show me your `docker-compose.yml` ?

**Comment by amgarr** [comment]
I have gotten it running but I don't think I have the `APP_BASE_URL` set up correctly. I got postgres runningby adjusting `pg_hba.conf` My `docker-compose.yml` is below. I can't access the API using the service address. 

```
version: '3.8'
services:
  waistline-api:
    image: waistline-api:latest
    container_name: waistline-api
    depends_on:
      - postgres
      - mongodb
    volumes:
      - .env.local:/var/www/html/.env.local
      - ./data/log:/var/log/apache2/ #optional
    ports:
      - "8081:80"
    healthcheck:
        test: ["CMD-SHELL", "wget -q --tries=1 http://localhost:80/api/v1/health -O - | grep -Eo '\"healthy\"\\s*:\\s*true' || exit 1"]
        interval: 60s
        timeout: 10s
        retries: 3
        start_period: 40s
    restart: unless-stopped

  postgres:
    image: postgres:16.1
    container_name: waistline-api_postgres
    environment:
      POSTGRES_DB: 'waistline'
      POSTGRES_USER: 'app'
      POSTGRES_PASSWORD: '<password>'
    volumes:
      - ./data/postgres:/var/lib/postgresql/data
    restart: unless-stopped

  mongodb:
    image: mongo:7.0
    container_name: waistline-api_mongodb
    environment:
      MONGO_INITDB_ROOT_USERNAME: 'root'
      MONGO_INITDB_ROOT_PASSWORD: '<password>'
    volumes:
      - ./data/mongodb:/data/db
    restart: unless-stopped
```

**Comment by hippothomas** [comment]
I tried your configuration where I just changed the image source to `waistlineapi/waistline-api:latest` and it works fine for me.

Here is the `.env.local` file I used :
```dotenv
APP_ENV=prod
APP_SECRET=03ed5a371f9306722ca7c1b9d22d92bd

DATABASE_URL="postgresql://app:password@waistline-api_postgres:5432/waistline?serverVersion=16&charset=utf8"

APP_BASE_URL="http://127.0.0.1:8081"
APP_REGISTER=true
API_DAILY_LIMIT=500
MONGODB_URL="mongodb://root:password@waistline-api_mongodb:27017"
MONGODB_NAME="waistline-api"

SENTRY_DSN=
```

Then I started the docker compose & initialized the db with:
```bash
docker compose up -d
docker exec -it waistline-api /bin/sh
php bin/console doctrine:migrations:migrate --quiet
php bin/console cache:clear --quiet
exit
```

Once the app was started, I went to `http://127.0.0.1:8081/` where I could login/register and get my API Key.
To use the API, I used curl with this command:
```bash
curl 'http://127.0.0.1:8081/api/v1/stats' -H "Authorization: ApiKey 018df6a1-7e65-72b3-9fd6-2b7a698ccbaa"
```
And got this result:
```json
{"pagination":{"limit":100,"offset":0,"total":0,"count":0},"results":[]}
```

Is your app deployed on a server? In this case do you use a reverse proxy to serve the app?

ps : In the config the `APP_BASE_URL` is not really important, it's just used in the dashboard to show the api url but if you put something wrong it will not affect the app.

**Comment by amgarr** [comment]
Huh, I actually got it working. I had the app secret in `""`. I think that was my main issue. 

API is now working but I am having trouble getting Waistline to send the data. Do I need to set up a waistline-api behind reverse proxy so Waistline can access it?

I have been deploying the app on my computer. Don't have reverse proxy set up. 

Edit: Never mind. I think I got it all sorted. Thanks for your help!
